### PR TITLE
server: add ServerService.serviceFindOrCreate

### PR DESCRIPTION
**Pull request by sam-github**
These methods were already available from Client, but the pm needs to
call them as well.

connected to strongloop-internal/scrum-nodeops#511

### CODE REVIEW

File: client/models/server-service.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ module.exports = function extServerService(ServerService) {
         var instance = instances[i];
         if (!instance.containerVersionInfo)
           continue;
-        var container = instance.containerVersionInfo.container;
+        var container = instance.containerVersionInfo.container; //
```
Comment (rmg) [inline client/models/server-service.js]: typo comment?

File: common/models/api.json [hunk 2]
```diff
@@ -32,7 +32,8 @@
     "driverStatus": {
       "type": "string",
       "required": true,
-      "description": "The hostname of this executor"
+      "XXX": "value is always 'started', what is it for?",
```
Comment (rmg) [inline common/models/api.json]: I think there's a convention of `_comment` as the key for things like this.

File: server/models/server-service.js [hunk 3]
```diff
@@ -70,6 +70,52 @@ module.exports = function extendServerService(ServerService) {
...
+    self.serviceFind(name, function(err, service) {
+      if (err && err.statusCode !== 404) return callback(err);
+      if (service) return callback(null, service);
+      self.serviceCreate(name, scale, callback);
+    });
+  }
+  ServerService.serviceFindOrCreate = serviceFindOrCreate;
+
+  function serviceCreate(name, scale, callback) {
+    var self = this;
```
Comment (rmg) [inline server/models/server-service.js]: You alias `this` twice.. `Service` makes sense as a "commenting variable", but `self` is just redundant.

File: server/models/server-service.js [hunk 4]
```diff
@@ -70,6 +70,52 @@ module.exports = function extendServerService(ServerService) {
...
+    if (scale == null) scale = 1;
+    var svc = {name: name, _groups: [{id: 1, name: 'default', scale: scale}]};
+    Service.create(svc, callback);
+
+    return self;
+  }
+  ServerService.serviceCreate = serviceCreate;
+
+  function serviceFind(serviceNameOrId, callback) {
+    var self = this;
```
Comment (rmg) [inline server/models/server-service.js]: Ditto here regarding aliasing of `this`.

**Comment by rmg** [comment]
I half expected based on the PR description to see these methods added to common/models and removed from the client...

Actually, in both cases I'm not sure why they are needed at all.. it isn't much of an abstraction over:
```js
var filter = {where: {name: name}};
var service = {name: name, _groups: [{id: 1, name: 'default', scale: scale}]};
Service.findOrCreate(filter, service, callback);
```

**Comment by rmg** [comment]
Other than the comment and `this` aliasing nits, LGTM to merge.

File: common/models/api.json (hunk 2 above)
Comment (sam-github) [inline common/models/api.json]: I committed a bunch of temp changes, I'm sorry.

File: server/models/server-service.js (hunk 3 above)
Comment (sam-github) [inline server/models/server-service.js]: keeps lint happy, but I'll add an anti-lint comment and simplify the code. lint hates `Service = this`, it expects all this to be assigned to self...

**Comment by sam-github** [comment]
re: https://github.com/strongloop/strong-mesh-models/pull/42#issuecomment-97894086 I think that might work, but then it would change the client API, and I don't want to break that.

`Service.findOrCreate` <--- no such method exists, that's what this PR introduces

**Comment by rmg** [comment]
Is `Service` not a `PersistedModel`?

**Pull request closed without merging**